enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,348 @@
+}
+
+// LoadBalancerDNSManagementPolicy is a policy for configuring how
+// ingresscontrollers manage DNS.
+//
+// +kubebuilder:validation:Enum=Managed;Unmanaged
+type LoadBalancerDNSManagementPolicy string
+
+const (
+    // ManagedLoadBalancerDNS configures the ingresscontroller to manage
+    // the wildcard DNS records associated with it.
+    ManagedLoadBalancerDNS LoadBalancerDNSManagementPolicy = "Managed"
+    // UnmanagedLoadBalancerDNS configures the ingresscontroller to not
+    // manage the wildcard DNS records associated with it.
+    UnmanagedLoadBalancerDNS LoadBalancerDNSManagementPolicy = "Unmanaged"
+)
+
+// DNSRecordSpec contains the details of a DNS record.
+type DNSRecordSpec struct {
+  // <snip>
+
+  // dnsManagementPolicy denotes the current policy applied on the dns record.
+  // Records that have policy set as "Unmanaged" are ignored by the ingress
+  // operator and should be managed at the discretion of the cluster admin.
+  //
+  // +kubebuilder:default:="Managed"
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:Enum=Managed;Unmanaged
+  DNSManagementPolicy DNSManagementPolicy `json:"dnsManagementPolicy"`
+}
+
+// DNSManagementPolicy is a policy for configuring how the dns controller 
+// manages DNSRecords.
+//
+// +kubebuilder:validation:Enum=Managed;Unmanaged
+type DNSManagementPolicy string
+
+const (
+    // ManagedDNS configures the dns controller manage the lifecyle of the
+    // dns record on the appropriate platform.
+    ManagedDNS DNSManagementPolicy = "Managed"
+    // UnmanagedDNS configures the dns controller to ignore the dns record
+    // and allows to be manually managed by the cluster admin.
+    UnmanagedDNS DNSManagementPolicy = "Unmanaged"
+)
+```
+
+Re-uses existing condition `DNSManaged` of the controller to denote if
+the wildcard *DNSRecord* associated with the controller is managed/unmanaged. 
+
+### Implementation Details/Notes/Constraints
+
+Based on `.loadbalancer.dnsManagementPolicy`, the controller decides whether to ensure creating
+a wildcard *DNSRecord*. If the customer is updating `.loadbalancer.dnsManagementPolicy` from 
+`"Managed"` to `"Unmanaged"` the previously created *DNSRecord* will be updated
+appropriately.
+
+The existing `.loadbalancer.dnsManagementPolicy` value of the ingresscontroller
+will be replicated to the *DNSRecord*'s `.spec.dnsManagementPolicy` as well, to
+clearly state which dns record is unmanaged. The cluster admin can choose to delete
+the *DNSRecord* at his own discretion (this will involve also manually removing the finalizer).
+
+Once the *DNSRecord* is set to `Unmanaged`, any updates done to it will be ignored
+by the operator, but it is to be noted that if the ingresscontroller is updated from
+`Unmanaged` to `Managed` any changes done will be lost during re-sync.
+
+If the ingresscontroller is deleted, and the associated *DNSRecord* still exists
```

Comment:
Clarify the implementation details on how deletion of ingresscontrollers will be done.

Reply:
I know that there is probably no easy way out. But I think it still a bit contradictory. In [this comment](https://github.com/openshift/enhancements/pull/1114/files#r886931980) we say that `DNSRecord` CR won't be created/updated/deleted, however the deletion will still happen: `deletionTimestamp` will be added. Also the same comment says that it's the full responsibility of the cluster admin to deal with `DNSRecord` CRs however the cluster admin won't do `oc delete dnsrecord` it has to be `oc edit dnsrecord` which is kinda not intuitive.

I think I'm rather for one of the following:
- "fully orphaned" unmanaged `DNSRecord` CR: cluster admin has to do `aws route53 delete resource record` + `oc delete dnsrecord`. The orphaned `DNSRecord` CRs can be found with [the command you suggested](https://github.com/openshift/enhancements/pull/1114/files#r881753249).
- "cloud provider orphaned" unmanaged `DNSRecord` CR: cluster admin has to do `aws route53 delete resource record` only. `DNSRecord` will be cleared from the finalization label without doing the actual finalization.

@thejasn, @brandisher: WDYT guys?